packages/core/src/codecatalyst/devEnv.ts

```diff
@@ -53,67 +49,69 @@ export class DevEnvActivityStarter {
         return waitUntil(
             async () => {
                 await this.tryStartDevEnvActivityHeartbeat()
-                return this.didStart
+                return !!this.devEnvActivity
             },
             { interval: 20_000, timeout: 60_000 * 5 }
         )
     }
 
     /** Trys to start the Activity Heartbeat mechanism */
-    private tryStartDevEnvActivityHeartbeat = shared(async () => {
-        if (this.didStart) {
+    private tryStartDevEnvActivityHeartbeat = shared(async (reauth: boolean = false) => {
+        if (!!this.devEnvActivity && !reauth) {
             return
         }
 
         const thisDevenv = (await getThisDevEnv(this.authProvider))?.unwrapOrElse(err => {
             getLogger().warn('codecatalyst: failed to get current Dev Enviroment: %s', err)
+            getLogger().warn('codecatalyst: initiating activity heartbeat, assuming 5 minute timeout')
             return undefined
         })
 
-        if (!thisDevenv) {
+        if (!thisDevenv && reauth) {
+            getLogger().error('codecatalyst: failed to get dev env after reauthenticate attempt')
+            return
+        } else if (!thisDevenv) {
             const connection = this.authProvider.activeConnection
             if (connection) {
                 void vscode.window
-                    .showErrorMessage(
-                        'CodeCatalyst: Reauthenticate your connection or the Dev Environment will time out.',
-                        'Reauthenticate'
-                    )
+                    .showErrorMessage('CodeCatalyst: Reauthenticate your connection.', 'Reauthenticate')
                     .then(async res => {
                         if (res !== 'Reauthenticate') {
                             return
                         }
                         await this.authProvider.auth.reauthenticate(connection)
-                        void this.tryStartDevEnvActivityHeartbeat()
+                        void this.tryStartDevEnvActivityHeartbeat(true)
                     })
-
-                getLogger().warn('codecatalyst: dev env needs reauth to not time out')
-            } else {
-                getLogger().warn(`codecatalyst: dev env needs a connection to not time out`)
             }
-            return
         }
 
-        const maxInactivityMinutes = thisDevenv.summary.inactivityTimeoutMinutes
-        if (!shouldTrackUserActivity(maxInactivityMinutes)) {
-            getLogger().debug(
-                `codecatalyst: not tracking user inactivity due to inactivity minutes being: ${maxInactivityMinutes}`
+        // If user is not authenticated, assume 5 minutes.
+        const inactivityTimeoutMin = thisDevenv?.summary.inactivityTimeoutMinutes ?? 5
```

Comment:
This was the only reason we needed the getThisDevEnv() result. But we can assume 5 minutes in the worst case.